sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -41,186 +42,433 @@ enum EntityType {
+            await recreateTopic(managementTopic1);
+            await recreateSubscription(managementTopic1, managementSubscription1);
+            await recreateSubscription(managementTopic1, managementSubscription2);
+            break;
+
+          case EntityType.RULE:
+            await recreateTopic(managementTopic1);
+            await recreateSubscription(managementTopic1, managementSubscription1);
+            await createEntity(
+              EntityType.RULE,
+              managementRule1,
+              managementTopic1,
+              managementSubscription1
+            );
+            await createEntity(
+              EntityType.RULE,
+              managementRule2,
+              managementTopic1,
+              managementSubscription1
+            );
+            break;
+          default:
+            throw new Error("TestError: Unrecognized EntityType");
+        }
+      });
+
+      afterEach(async () => {
+        switch (entityType) {
+          case EntityType.QUEUE:
+            await deleteEntity(EntityType.QUEUE, managementQueue1);
+            await deleteEntity(EntityType.QUEUE, managementQueue2);
+            break;
+
+          case EntityType.TOPIC:
+            await deleteEntity(EntityType.TOPIC, managementTopic1);
+            await deleteEntity(EntityType.TOPIC, managementTopic2);
+            break;
+          case EntityType.SUBSCRIPTION:
+          case EntityType.RULE:
+            await deleteEntity(EntityType.TOPIC, managementTopic1);
+            break;
+
+          default:
+            throw new Error("TestError: Unrecognized EntityType");
+        }
+      });
+
+      it(`List on existing entities for type ${entityType} with top 1 returns the first entity`, async () => {
+        const allEntities = await listEntities(
+          entityType,
+          managementTopic1,
+          managementSubscription1
+        );
+
+        const topOneEntity = await listEntities(
+          entityType,
+          managementTopic1,
+          managementSubscription1,
+          undefined,
+          1
+        );
+
+        should.equal(
+          Array.isArray(topOneEntity),
+          true,
+          "Result must be any array for list requests"
+        );
+        should.equal(topOneEntity.length, 1, "Result must be an empty array");
+        should.equal(
+          allEntities[0][entityType.toLowerCase() + "Name"],
+          topOneEntity[0][entityType.toLowerCase() + "Name"],
+          "Entity name mismatch"
+        );
+      });
+
+      it(`List on existing entities for type ${entityType} with skip 1 returns all entities skipping 1`, async () => {
+        const skipEntitiesResult = await listEntities(
+          entityType,
+          managementTopic1,
+          managementSubscription1,
+          1,
+          undefined
+        );
+
+        should.equal(
+          Array.isArray(skipEntitiesResult),
+          true,
+          "Result must be any array for list requests"
```

Comment:
The checks here only look for the result of the list operation being an array. But the test title suggests more.

Reply:
I recall discussing about this with @AlexGhiondea 
We wanted to lax on this check and have the test to make a successful request to server.
We contemplated on having a unit test that ensures the request is getting constructed correctly, however we landed on having a live test in place to ensure the request is made and the service responds with a valid result.
100% accuracy is not possible as the number of entities in the namespace keep changing during the test runs.

Reply:
> 100% accuracy is not possible as the number of entities in the namespace keep changing during the test runs.

That makes sense as we now have the node and the browser test runs running in parallel using the same namespace. But, once we start using the [ARM template to create a Service Bus namespace for each test run](https://github.com/Azure/azure-sdk-for-js/issues/6605#issuecomment-567305888), we won't have this problem. 

Please create an issue to update these tests once we have the ARM templates in place. Mark that new issue as blocked by #6605

Reply:
> 100% accuracy is not possible as the number of entities in the namespace keep changing during the test runs.

That makes sense as we now have the node and the browser test runs running in parallel using the same namespace. But, once we start using the [ARM template to create a Service Bus namespace for each test run](https://github.com/Azure/azure-sdk-for-js/issues/6605#issuecomment-567305888), we won't have this problem. 

Please create an issue to update these tests once we have the ARM templates in place. Mark that new issue as blocked by #6605

Reply:
> 100% accuracy is not possible as the number of entities in the namespace keep changing during the test runs.

That makes sense as we now have the node and the browser test runs running in parallel using the same namespace. But, once we start using the [ARM template to create a Service Bus namespace for each test run](https://github.com/Azure/azure-sdk-for-js/issues/6605#issuecomment-567305888), we won't have this problem. 

Please create an issue to update these tests once we have the ARM templates in place. Mark that new issue as blocked by #6605

Reply:
Just a thought - seems very heavy weight to have to recreate namespaces for every test run. Having it as an optional utility for testcases as these seems pretty useful though. 

Since #6605 title is "update all test cases", we should be able to re-use it to update this test-case. In general, all of the test cases will need to be closely looked at updated anyway.
Posted a comment on it to capture the same.

Reply:
> seems very heavy weight to have to recreate namespaces for every test run. Having it as an optional utility for testcases as these seems pretty useful though.

Here "test run" means the entire test run say for node or browser and not each test case. So, not heavy weight. This is in compliance with the test guidelines where we want each test run to use its own resource and delete it in the end

> Since #6605 title is "update all test cases", we should be able to re-use it to update this test-case. 

I have closed #6605 in favor of #6650 as it had turned to only the arm template related work.

Please create a new issue for improving the list test cases and mark it as blocked by #6650

Reply:
Discussed offline - filed https://github.com/Azure/azure-sdk-for-js/issues/6723 since #6605 has been updated

Reply:
> seems very heavy weight to have to recreate namespaces for every test run. Having it as an optional utility for testcases as these seems pretty useful though.

Here "test run" means the entire test run say for node or browser and not each test case. So, not heavy weight. This is in compliance with the test guidelines where we want each test run to use its own resource and delete it in the end

> Since #6605 title is "update all test cases", we should be able to re-use it to update this test-case. 

I have closed #6605 in favor of #6650 as it had turned to only the arm template related work.

Please create a new issue for improving the list test cases and mark it as blocked by #6650